src/main/groovy/com/morpheusdata/upcloud/UpcloudCloudProvider.groovy

```diff
@@ -140,7 +164,67 @@ class UpcloudCloudProvider implements CloudProvider {
 	 */
 	@Override
 	Collection<ComputeServerType> getComputeServerTypes() {
-		Collection<ComputeServerType> serverTypes = []
+		Collection<ComputeServerType> serverTypes = [
+			new ComputeServerType(
+				meta:[key:'code'], code:'selfManagedLinux', name:'Manual Docker Host', description:'', platform:'linux', nodeType:'morpheus-node',
+				enabled:true, selectable:false, externalDelete:false, managed:true, controlPower:false, controlSuspend:false, creatable:true, computeService:'standardComputeService',
+				deleteQueue:'computeStandardDeleteServerQueue', initializeQueue:'computeStandardInitializeServerQueue', displayOrder:16, hasAutomation:true,
+				hasAgent:true, containerHypervisor:true, bareMetalHost:false, vmHypervisor:false, agentType:'node', containerEngine:'docker',
+				provisionType:[domainClass:'provisionType', meta:[key:'code'], code:'manual'], canSyncHypervisorStats: false,
+				computeType:[domainClass:'computeType', meta:[key:'code'], code:'docker-host'],
+				morpheusHypervisorType: [domainClass: 'computeServerType', meta: [key: 'code'], code: 'morpheusComboLinux'],
+				optionTypes:[
+					[meta:[key:'code'], code:'computeServerType.global.sshHost'],
```

Comment:
these will need to be converted to OptionType objects, but they can still just reference the code.